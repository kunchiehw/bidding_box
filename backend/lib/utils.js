const jwt = require('jsonwebtoken');

const secret = process.env.SHARE_SECRET;
const db = {
  users: {
    wkc: 'password',
    jarron: 'password',
  },
};


function authenticateUser(username, password) {
  return new Promise((resolve, reject) => {
    if (!username || !password || !(username in db.users) || db.users[username] !== password) {
      return reject();
    }

    const token = jwt.sign({ username }, secret, { expiresIn: '30d' });
    return resolve(token);
  });
}
module.exports.authenticateUser = authenticateUser;


function validateJwt(token) {
  return new Promise((resolve, reject) => {
    jwt.verify(token, secret, (err, decoded) => {
      if (err || !decoded) {
        reject();
      }

      if (!('username' in decoded) || !(decoded.username in db.users)) {
        reject();
      }

      resolve(token);
    });
  });
}
module.exports.validateJwt = validateJwt;


module.exports.validateJwtMiddleware = (req, res, next) => {
  if (req.headers.authorization && req.headers.authorization.split(' ')[0] === 'Bearer') {
    const token = req.headers.authorization.split(' ')[1];
    validateJwt(token)
      .then(() => next())
      .catch(() => next(new Error('Unauthorized')));
  } else {
    next(new Error('Unauthorized'));
  }
};


module.exports.getTtl = () =>
  Math.floor(Date.now() / 1000) + (4 * 60 * 60); // ttl for 4 hour
