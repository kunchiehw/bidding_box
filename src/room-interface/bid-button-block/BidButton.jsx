import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Button } from 'semantic-ui-react';
import { bidToString, suitColor } from '../helper-RoomInterface';
import { ALL_SUITS } from '../../util/util';

const propTypes = {
  // TODO: Might need a specific validator
  level: PropTypes.number.isRequired,
  suit: PropTypes.oneOf(ALL_SUITS).isRequired,
  isDisabled: PropTypes.bool.isRequired,
  handleBidButtonClick: PropTypes.func.isRequired,
};

class BidButton extends PureComponent {
  render() {
    return (
      <Button
        className="bid-button"
        onClick={() => { this.props.handleBidButtonClick(this.props.level, this.props.suit); }}
        disabled={this.props.isDisabled}
        color={suitColor(this.props.suit)}
        size="small"
      >
        {bidToString(this.props.level, this.props.suit)}
      </Button>
    );
  }
}

BidButton.propTypes = propTypes;

export default BidButton;
