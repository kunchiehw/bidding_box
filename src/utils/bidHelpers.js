const { STANDARD_SUITS } = require('./constants');

function isPass(bid) {
  return (bid && bid.suit === 'PASS');
}

function isDouble(bid) {
  return (bid && bid.suit === 'DOUBLE');
}

function isSuit(bid) {
  return (bid && STANDARD_SUITS.includes(bid.suit));
}

function isBidSeqEnded(bidSeq) {
  const bidSeqLen = bidSeq.length;
  if (isPass(bidSeq[bidSeqLen - 3]) &&
      isPass(bidSeq[bidSeqLen - 2]) &&
      isPass(bidSeq[bidSeqLen - 1])) {
    return (bidSeqLen >= 4);
  }
  return false;
}

function isDoubleAllowed(bidSeq) {
  const bidSeqLen = bidSeq.length;
  if (isSuit(bidSeq[bidSeqLen - 1])) {
    return false;
  }
  if (isSuit(bidSeq[bidSeqLen - 3]) &&
      isPass(bidSeq[bidSeqLen - 2]) &&
      isPass(bidSeq[bidSeqLen - 1])) {
    return false;
  }
  return true;
}

function isRedoubleAllowed(bidSeq) {
  const bidSeqLen = bidSeq.length;
  if (isDouble(bidSeq[bidSeqLen - 1])) {
    return false;
  }
  if (isDouble(bidSeq[bidSeqLen - 3]) &&
      isPass(bidSeq[bidSeqLen - 2]) &&
      isPass(bidSeq[bidSeqLen - 1])) {
    return false;
  }
  return true;
}

function getCurrentBid(bidSeq) {
  for (let i = bidSeq.length - 1; i >= 0; i -= 1) {
    if (isSuit(bidSeq[i])) {
      return bidSeq[i];
    }
  }
  return { level: 0, suit: 'PASS' };
}

function isStandardBidAllowed(currentLevel, currentSuit, level, suit) {
  if (currentLevel !== level) {
    return (currentLevel > level);
  }
  return (STANDARD_SUITS.indexOf(currentSuit) >= STANDARD_SUITS.indexOf(suit));
}


module.exports = {
  isPass,
  isDouble,
  isSuit,
  isBidSeqEnded,
  isDoubleAllowed,
  isRedoubleAllowed,
  getCurrentBid,
  isStandardBidAllowed,
};
