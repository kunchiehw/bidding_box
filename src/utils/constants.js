export const POSSIBLE_LEVELS = [1, 2, 3, 4, 5, 6, 7];

export const STANDARD_SUITS = ['CLUBS', 'DIAMONDS', 'HEARTS', 'SPADES', 'NOTRUMPS'];
export const SPECIAL_SUITS = ['PASS', 'DOUBLE', 'REDOUBLE'];
export const ALL_SUITS = ['CLUBS', 'DIAMONDS', 'HEARTS', 'SPADES', 'NOTRUMPS', 'PASS', 'DOUBLE', 'REDOUBLE'];

export const NS_SEATS = ['NORTH', 'SOUTH'];
export const EW_SEATS = ['WEST', 'EAST'];
export const SEATS = ['WEST', 'NORTH', 'EAST', 'SOUTH'];
export const DECLARERS = ['WEST', 'NORTH', 'EAST', 'SOUTH', 'NS', 'EW'];
export const PARTICIPANTS = ['WEST', 'EAST', 'WATCHER'];

export const VULNERABILITIES = ['NONE', 'NS', 'EW', 'BOTH'];

export const BOOLEANS = [true, false];
