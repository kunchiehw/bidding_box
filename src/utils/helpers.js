import { STANDARD_SUITS } from './constants';
import getMessage from '../i18n/messages';

export function bidToString(level, suit) {
  if (STANDARD_SUITS.includes(suit)) {
    return `${level} ${getMessage(suit)}`;
  }
  return getMessage(suit);
}

export function suitColor(suit) {
  switch (suit) {
    case 'NOTRUMPS':
      return 'black';
    case 'SPADES':
    case 'REDOUBLE':
      return 'blue';
    case 'HEARTS':
    case 'DOUBLE':
      return 'red';
    case 'DIAMONDS':
      return 'orange';
    case 'CLUBS':
    case 'PASS':
      return 'green';
    default:
      return '';
  }
}

export function shouldStandardBidButtonDisabled(currentLevel, currentSuit, buttonLevel, buttonSuit) {
  if (currentLevel !== buttonLevel) {
    return (currentLevel > buttonLevel);
  }
  return (STANDARD_SUITS.indexOf(currentSuit) >= STANDARD_SUITS.indexOf(buttonSuit));
}
