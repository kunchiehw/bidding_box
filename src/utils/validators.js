import { POSSIBLE_LEVELS, STANDARD_SUITS } from './constants';

// HOC for all validator
function createChainableTypeChecker(validate) {
  function checkType(isRequired, props, propName, componentName, location) {
    componentName = componentName || 'ANONYMOUS';
    if (props[propName] == null) {
      if (isRequired) {
        return new Error(`'${componentName}': requires prop '${propName}'.`);
      }
      return null;
    }
    return validate(props, propName, componentName, location);
  }

  const chainedCheckType = checkType.bind(null, false);
  chainedCheckType.isRequired = checkType.bind(null, true);

  return chainedCheckType;
}

export const levelValidator = createChainableTypeChecker((props, propName, componentName) => {
  componentName = componentName || 'ANONYMOUS';

  if (props.suit == null) {
    return new Error(`'${componentName}': requires prop 'suit' when validates prop '${propName}'.`);
  }

  if (props[propName]) {
    const level = props[propName];
    if (STANDARD_SUITS.includes(props.suit)) {
      if (!POSSIBLE_LEVELS.includes(level)) {
        return new Error(`'${componentName}': '${propName}' should be in POSSIBLE_LEVELS when 'suit' is in STANDARD_SUITS.`);
      }
    } else if (level !== 0) {
      return new Error(`'${componentName}': '${propName}' should be 0 when 'suit' is in SPECIAL_SUITS.`);
    }
  }

  return null;
});

export function foo() {}
