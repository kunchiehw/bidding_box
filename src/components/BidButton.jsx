import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Button } from 'semantic-ui-react';
import { ALL_SUITS } from '../utils/constants';
import { bidToString, suitColor } from '../utils/helpers';
import { levelValidator } from '../utils/validators';

const propTypes = {
  level: levelValidator.isRequired,
  suit: PropTypes.oneOf(ALL_SUITS).isRequired,
  isDisabled: PropTypes.bool.isRequired,
  handleBidButtonClick: PropTypes.func.isRequired,
};

class BidButton extends PureComponent {
  render() {
    return (
      <Button
        className="bid-button"
        onClick={() => { this.props.handleBidButtonClick(this.props.level, this.props.suit); }}
        disabled={this.props.isDisabled}
        color={suitColor(this.props.suit)}
        size="small"
      >
        {bidToString(this.props.level, this.props.suit)}
      </Button>
    );
  }
}

BidButton.propTypes = propTypes;

export default BidButton;
