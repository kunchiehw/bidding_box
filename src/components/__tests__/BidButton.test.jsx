import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import BidButton from '../BidButton';
import { POSSIBLE_LEVELS, STANDARD_SUITS, SPECIAL_SUITS, BOOLEAN_CHOICES } from '../../utils/constants';

describe('BidButton render', () => {
  POSSIBLE_LEVELS.forEach((level) => {
    STANDARD_SUITS.forEach((suit) => {
      BOOLEAN_CHOICES.forEach((isDisabled) => {
        it(`level: ${level}; suit: ${suit}, isDisabled: ${isDisabled}`, () => {
          const bidButton = shallow(<BidButton
            level={level}
            suit={suit}
            isDisabled={isDisabled}
            handleBidButtonClick={() => null}
          />);
          expect(toJson(bidButton)).toMatchSnapshot();
        });
      });
    });
  });

  SPECIAL_SUITS.forEach((suit) => {
    BOOLEAN_CHOICES.forEach((isDisabled) => {
      it(`suit: ${suit}, isDisabled: ${isDisabled}`, () => {
        const bidButton = shallow(<BidButton
          level={0}
          suit={suit}
          isDisabled={isDisabled}
          handleBidButtonClick={() => null}
        />);
        expect(toJson(bidButton)).toMatchSnapshot();
      });
    });
  });
});
