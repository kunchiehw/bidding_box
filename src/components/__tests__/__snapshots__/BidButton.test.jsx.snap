// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BidButton render level: 1; suit: CLUBS, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="green"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  1 ♣
</Button>
`;

exports[`BidButton render level: 1; suit: CLUBS, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="green"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  1 ♣
</Button>
`;

exports[`BidButton render level: 1; suit: DIAMONDS, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="orange"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  1 ♦
</Button>
`;

exports[`BidButton render level: 1; suit: DIAMONDS, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="orange"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  1 ♦
</Button>
`;

exports[`BidButton render level: 1; suit: HEARTS, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="red"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  1 ♥
</Button>
`;

exports[`BidButton render level: 1; suit: HEARTS, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="red"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  1 ♥
</Button>
`;

exports[`BidButton render level: 1; suit: NOTRUMPS, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="black"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  1 N
</Button>
`;

exports[`BidButton render level: 1; suit: NOTRUMPS, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="black"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  1 N
</Button>
`;

exports[`BidButton render level: 1; suit: SPADES, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="blue"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  1 ♠
</Button>
`;

exports[`BidButton render level: 1; suit: SPADES, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="blue"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  1 ♠
</Button>
`;

exports[`BidButton render level: 2; suit: CLUBS, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="green"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  2 ♣
</Button>
`;

exports[`BidButton render level: 2; suit: CLUBS, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="green"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  2 ♣
</Button>
`;

exports[`BidButton render level: 2; suit: DIAMONDS, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="orange"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  2 ♦
</Button>
`;

exports[`BidButton render level: 2; suit: DIAMONDS, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="orange"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  2 ♦
</Button>
`;

exports[`BidButton render level: 2; suit: HEARTS, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="red"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  2 ♥
</Button>
`;

exports[`BidButton render level: 2; suit: HEARTS, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="red"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  2 ♥
</Button>
`;

exports[`BidButton render level: 2; suit: NOTRUMPS, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="black"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  2 N
</Button>
`;

exports[`BidButton render level: 2; suit: NOTRUMPS, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="black"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  2 N
</Button>
`;

exports[`BidButton render level: 2; suit: SPADES, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="blue"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  2 ♠
</Button>
`;

exports[`BidButton render level: 2; suit: SPADES, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="blue"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  2 ♠
</Button>
`;

exports[`BidButton render level: 3; suit: CLUBS, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="green"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  3 ♣
</Button>
`;

exports[`BidButton render level: 3; suit: CLUBS, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="green"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  3 ♣
</Button>
`;

exports[`BidButton render level: 3; suit: DIAMONDS, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="orange"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  3 ♦
</Button>
`;

exports[`BidButton render level: 3; suit: DIAMONDS, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="orange"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  3 ♦
</Button>
`;

exports[`BidButton render level: 3; suit: HEARTS, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="red"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  3 ♥
</Button>
`;

exports[`BidButton render level: 3; suit: HEARTS, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="red"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  3 ♥
</Button>
`;

exports[`BidButton render level: 3; suit: NOTRUMPS, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="black"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  3 N
</Button>
`;

exports[`BidButton render level: 3; suit: NOTRUMPS, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="black"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  3 N
</Button>
`;

exports[`BidButton render level: 3; suit: SPADES, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="blue"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  3 ♠
</Button>
`;

exports[`BidButton render level: 3; suit: SPADES, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="blue"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  3 ♠
</Button>
`;

exports[`BidButton render level: 4; suit: CLUBS, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="green"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  4 ♣
</Button>
`;

exports[`BidButton render level: 4; suit: CLUBS, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="green"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  4 ♣
</Button>
`;

exports[`BidButton render level: 4; suit: DIAMONDS, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="orange"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  4 ♦
</Button>
`;

exports[`BidButton render level: 4; suit: DIAMONDS, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="orange"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  4 ♦
</Button>
`;

exports[`BidButton render level: 4; suit: HEARTS, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="red"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  4 ♥
</Button>
`;

exports[`BidButton render level: 4; suit: HEARTS, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="red"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  4 ♥
</Button>
`;

exports[`BidButton render level: 4; suit: NOTRUMPS, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="black"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  4 N
</Button>
`;

exports[`BidButton render level: 4; suit: NOTRUMPS, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="black"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  4 N
</Button>
`;

exports[`BidButton render level: 4; suit: SPADES, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="blue"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  4 ♠
</Button>
`;

exports[`BidButton render level: 4; suit: SPADES, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="blue"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  4 ♠
</Button>
`;

exports[`BidButton render level: 5; suit: CLUBS, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="green"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  5 ♣
</Button>
`;

exports[`BidButton render level: 5; suit: CLUBS, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="green"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  5 ♣
</Button>
`;

exports[`BidButton render level: 5; suit: DIAMONDS, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="orange"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  5 ♦
</Button>
`;

exports[`BidButton render level: 5; suit: DIAMONDS, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="orange"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  5 ♦
</Button>
`;

exports[`BidButton render level: 5; suit: HEARTS, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="red"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  5 ♥
</Button>
`;

exports[`BidButton render level: 5; suit: HEARTS, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="red"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  5 ♥
</Button>
`;

exports[`BidButton render level: 5; suit: NOTRUMPS, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="black"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  5 N
</Button>
`;

exports[`BidButton render level: 5; suit: NOTRUMPS, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="black"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  5 N
</Button>
`;

exports[`BidButton render level: 5; suit: SPADES, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="blue"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  5 ♠
</Button>
`;

exports[`BidButton render level: 5; suit: SPADES, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="blue"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  5 ♠
</Button>
`;

exports[`BidButton render level: 6; suit: CLUBS, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="green"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  6 ♣
</Button>
`;

exports[`BidButton render level: 6; suit: CLUBS, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="green"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  6 ♣
</Button>
`;

exports[`BidButton render level: 6; suit: DIAMONDS, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="orange"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  6 ♦
</Button>
`;

exports[`BidButton render level: 6; suit: DIAMONDS, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="orange"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  6 ♦
</Button>
`;

exports[`BidButton render level: 6; suit: HEARTS, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="red"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  6 ♥
</Button>
`;

exports[`BidButton render level: 6; suit: HEARTS, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="red"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  6 ♥
</Button>
`;

exports[`BidButton render level: 6; suit: NOTRUMPS, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="black"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  6 N
</Button>
`;

exports[`BidButton render level: 6; suit: NOTRUMPS, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="black"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  6 N
</Button>
`;

exports[`BidButton render level: 6; suit: SPADES, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="blue"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  6 ♠
</Button>
`;

exports[`BidButton render level: 6; suit: SPADES, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="blue"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  6 ♠
</Button>
`;

exports[`BidButton render level: 7; suit: CLUBS, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="green"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  7 ♣
</Button>
`;

exports[`BidButton render level: 7; suit: CLUBS, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="green"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  7 ♣
</Button>
`;

exports[`BidButton render level: 7; suit: DIAMONDS, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="orange"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  7 ♦
</Button>
`;

exports[`BidButton render level: 7; suit: DIAMONDS, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="orange"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  7 ♦
</Button>
`;

exports[`BidButton render level: 7; suit: HEARTS, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="red"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  7 ♥
</Button>
`;

exports[`BidButton render level: 7; suit: HEARTS, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="red"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  7 ♥
</Button>
`;

exports[`BidButton render level: 7; suit: NOTRUMPS, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="black"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  7 N
</Button>
`;

exports[`BidButton render level: 7; suit: NOTRUMPS, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="black"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  7 N
</Button>
`;

exports[`BidButton render level: 7; suit: SPADES, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="blue"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  7 ♠
</Button>
`;

exports[`BidButton render level: 7; suit: SPADES, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="blue"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  7 ♠
</Button>
`;

exports[`BidButton render suit: DOUBLE, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="red"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  X
</Button>
`;

exports[`BidButton render suit: DOUBLE, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="red"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  X
</Button>
`;

exports[`BidButton render suit: PASS, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="green"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  PASS
</Button>
`;

exports[`BidButton render suit: PASS, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="green"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  PASS
</Button>
`;

exports[`BidButton render suit: REDOUBLE, isDisabled: false 1`] = `
<Button
  as="button"
  className="bid-button"
  color="blue"
  disabled={false}
  onClick={[Function]}
  size="small"
>
  XX
</Button>
`;

exports[`BidButton render suit: REDOUBLE, isDisabled: true 1`] = `
<Button
  as="button"
  className="bid-button"
  color="blue"
  disabled={true}
  onClick={[Function]}
  size="small"
>
  XX
</Button>
`;
